openpype/modules/ftrack/plugins/publish/collect_ftrack_family.py

```diff
@@ -34,68 +33,65 @@ def process(self, instance):
             self.log.warning("No profiles present for adding Ftrack family")
             return
 
-        add_ftrack_family = False
-        task_name = instance.data.get("task",
-                                      legacy_io.Session["AVALON_TASK"])
-        host_name = legacy_io.Session["AVALON_APP"]
+        host_name = instance.context.data["hostName"]
         family = instance.data["family"]
+        task_name = instance.data.get("task")
 
         filtering_criteria = {
             "hosts": host_name,
             "families": family,
             "tasks": task_name
         }
-        profile = filter_profiles(self.profiles, filtering_criteria,
-                                  logger=self.log)
+        profile = filter_profiles(
+            self.profiles,
+            filtering_criteria,
+            logger=self.log
+        )
+
+        add_ftrack_family = False
+        families = instance.data.setdefault("families", [])
 
         if profile:
-            families = instance.data.get("families")
             add_ftrack_family = profile["add_ftrack_family"]
-
             additional_filters = profile.get("advanced_filtering")
             if additional_filters:
-                self.log.info("'{}' families used for additional filtering".
-                              format(families))
+                families_s = set(families) | {family}
+                self.log.info(
+                    "'{}' families used for additional filtering".format(
+                        families_s))
                 add_ftrack_family = self._get_add_ftrack_f_from_addit_filters(
                     additional_filters,
-                    families,
+                    families_s,
                     add_ftrack_family
                 )
 
-            if add_ftrack_family:
-                self.log.debug("Adding ftrack family for '{}'".
-                               format(instance.data.get("family")))
-
-                if families:
-                    if "ftrack" not in families:
-                        instance.data["families"].append("ftrack")
-                else:
-                    instance.data["families"] = ["ftrack"]
+        result_str = "Not adding"
+        if add_ftrack_family:
+            result_str = "Adding"
+            self.log.debug("Adding ftrack family for '{}'".format(family))
```

Comment:
This shouldn't be here, imho. Line 75 handles both cases (addition/not addition).

Reply:
```suggestion
```